### PR TITLE
Add view tree renderer

### PR DESCRIPTION
**Pull request by apatronl**

### CODE REVIEW

File: redwood-tooling-codegen/src/main/kotlin/app/cash/redwood/tooling/codegen/types.kt [hunk 1]
```diff
@@ -70,7 +70,7 @@ internal object Redwood {
 internal object RedwoodTesting {
   val MutableWidget = ClassName("app.cash.redwood.compose.testing", "MutableWidget")
   val RedwoodTester = ClassName("app.cash.redwood.compose.testing", "RedwoodTester")
-  val ViewTree = ClassName("app.cash.redwood.compose.testing", "ViewTree")
+  val ViewTree = ClassName("app.cash.redwood.protocol", "ViewTree")
```
Comment (JakeWharton) [inline redwood-tooling-codegen/src/main/kotlin/app/cash/redwood/tooling/codegen/types.kt]: Can you move this into `RedwoodProtocol` object to maintain the association with the module the type comes from.

File: redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/ViewTreeRenderer.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+public class ViewTreeRenderer<W : Any>(
+  private val json: Json,
+) {
+  private val refuseAllEvents = EventSink { event ->
+    throw IllegalStateException("unexpected event: $event")
+  }
+
+  @Suppress("UNCHECKED_CAST")
+  public fun render(
+    view: TreehouseView,
```
Comment (JakeWharton) [inline redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/ViewTreeRenderer.kt:41]: I think eventually we want this to live in 'redwood-protocol-widget' and run directly on a `Widget.Children`, but landing it in here is a good starting location.

**Review by JakeWharton — APPROVED** [review]

File: redwood-tooling-codegen/src/main/kotlin/app/cash/redwood/tooling/codegen/types.kt (hunk 1 above)
Comment (swankjesse) [reply redwood-tooling-codegen/src/main/kotlin/app/cash/redwood/tooling/codegen/types.kt]: This confused me 'cause I was thinking you were talking about moving the ViewTree class somewhere else, but then I realized you just wanna move this line to follow line 57 in this same file.

File: redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/ViewTreeRenderer.kt (hunk 2 above)
Comment (swankjesse) [reply redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/ViewTreeRenderer.kt:41]: (If so, we’ll need a passed-in WidgetSystem. That’s a good trade-off in modularity though.)

**Review by swankjesse — APPROVED** [review]

**Pull request merged**